src/ntfsck.c

```diff
@@ -1162,6 +1162,49 @@ static int ntfsck_add_index_entry_orphaned_file(ntfs_volume *vol, s64 mft_no)
 	goto next_inode;
 }
 
+static void ntfsck_check_mft_record_unused(ntfs_volume *vol, s64 mft_num)
+{
+	MFT_RECORD *mrec;
+	u16 seq_no;
+	s64 pos = mft_num * vol->mft_record_size;
+	s64 count = 512;
+
+	mrec = ntfs_malloc(count);
+	if (!mrec) {
+		ntfs_log_perror("Couldn't allocate record %lld",
+				(long long)mft_num);
+		return;
+	}
+
+	if (ntfs_attr_pread(vol->mft_na, pos, count, mrec) != count) {
+		ntfs_log_perror("Couldn't read $MFT record %lld",
+				(long long)mft_num);
+		return;
+	}
+
+	if (!ntfs_is_file_record(mrec->magic) ||
+	    !(mrec->flags & MFT_RECORD_IN_USE)) {
+		ntfs_log_verbose("Record(%"PRId64") unused. Skipping.\n",
+				mft_num);
+		return;
+	}
+
+	ntfs_log_error("Record(%"PRId64") used. "
```

Comment:
do not use check_failed() and ntfsck_ask_repair() ?

Reply:
I've been thinking about it, but wouldn't it be possible without it? I think there will be too many logs.